packages/moda-sdk/src/profile/getName.ts

```diff
@@ -0,0 +1,38 @@
+import { Address } from "viem";
+import { AddressError, Config, ConfigError } from "../types";
+import { Result } from "../utils/types";
+import { verifyConfig } from "../utils/verifyConfig";
+import { isValidAddress } from "../utils/isValidAddress";
+import { ProfileABI } from "./abi/ProfileABI";
+import { ProfileError } from "./types";
+
+interface GetNameArgs {
+    config: Config;
+    profileAddress: Address;
+}
+
+export async function getName(
+    { config, profileAddress} : GetNameArgs,
+): Promise<Result<string, ConfigError | AddressError | ProfileError>> {
```

Comment:
ProfileError defines a lot of possible error messages, but most of these cannot be thrown in this function. We should only define errors that are returned.

Also, it would be better to define a single error type:

`export type NameError = ConfigError | AddressError | ...`;

Reply:
NameUndefined is not possible for the contract. It should always return a string